custom_metrics/almanac.js

```diff
@@ -306,6 +336,57 @@ return JSON.stringify({
       return null;
     }
   })(),
-  //  check if there is any picture tag containing an img tag
-  'has_picture_img': document.querySelectorAll('picture img').length > 0
+  // Counts the number of picture tags containing an img tag
+  'num_picture_img': document.querySelectorAll('picture img').length,
+  // Counts the number of source or img tags with sizes attribute
+  'num_image_sizes': document.querySelectorAll('source[sizes], img[sizes]').length,
+  // Count the mumber of images with srcset attribute
+  'num_srcset_all': document.querySelectorAll('source[srcset], img[srcset]').length,
+  // Count the mumber of images with srcset and sizes attributes
+  'num_srcset_sizes': document.querySelectorAll('source[srcset][sizes], img[srcset][sizes]').length,
+  // Count the number of imges with srcset with descriptor-x
+  'num_srcset_descriptor_x': document.querySelectorAll('source[srcset*="x"], img[srcset*="x"]').length,
+  // Count the number of imges with srcset with descriptor-w
+  'num_srcset_descriptor_w': document.querySelectorAll('source[srcset*="w"], img[srcset*="w"]').length,
+  // Count the number of scrset candidates
+  'num_srcset_candidates': (() => {
+    var nodes = document.querySelectorAll('source[srcset], img[srcset]');
+    var count = 0;
+    for (var i = 0, len = nodes.length; i < len; i++) {
+      var srcset = nodes[i]['srcset'];
+      count += srcset.split(',').length;
+    }
+    return count;
+  })(),
+  // Return picture formats ["image/webp","image/svg+xml"]
+  'picture_formats': (() => {
+    var nodes = document.querySelectorAll('picture source[type]');
+    var formats = {};
+    for (var i = 0, len = nodes.length; i < len; i++) {
+      formats[nodes[i]['type'].toLowerCase()] = '';
+    }
+    return Object.getOwnPropertyNames(formats);
+  })(),
+  // Count all video nodes
+  'num_video_nodes': document.querySelectorAll('video').length,
+  // Returns a set of video node attribute names
+  'video_nodes_attributes': (() => {
+    var allAttributes = getNodesAttributes(document.querySelectorAll('video'));
+    var filter = ['autoplay', 'autoPictureInPicture', 'buffered', 'controls',
+      'controlslist', 'crossorigin', 'use-credentials', 'currentTime',
+      'disablePictureInPicture', 'disableRemotePlayback', 'duration',
+      'height', 'intrinsicsize', 'loop', 'muted', 'playsinline', 'poster',
+      'preload', 'src', 'width'];
+    return allAttributes.filter(el => filter.includes(el));
+  })(),
+  // Counts the number of pictures using source media min-resolution
+  'num_picture_using_min_resolution': (() => {
+    var nodes = document.querySelectorAll('picture source[media*="min-resolution"]');
+    return countUniqueParents(nodes);
```

Comment:
May I suggest a simpler approach for this and `num_picture_using_orientation` below?

```suggestion
    var pictures = document.querySelectorAll('picture');
    return Array.from(pictures).filter(picture => {
      return picture.querySelector('source[media*="min-resolution"]');
    }).length;
```